fix: str_to_isoformat for unformatted datetime

0xAlcidius opened the pull request:
### Summary
Introduce `str_to_isoformat` to format datetime strings
encountered in Windows Task Scheduler XML. This ensures
non-padded months/days and space-separated datetimes are
converted into strict ISO 8601 strings.

### Motivation
Parsing a velo package could result in the following error:
```
  File "/usr/lib/pypy/lib/pypy3.11/datetime.py", line 1852, in fromisoformat
    raise ValueError(
ValueError: Invalid isoformat string: '2025-7-29T16:34:00'
```
This PR solves that issue.

bobkarreman commented:
Hi @0xAlcidius thanks for your contribution.

Can you please add some unit-tests for this case?

Maybe the fix can be much easier by parsing the datetime with datetime.strptime something like this:
```
datetime.strptime("2025-7-9T16:34:00", "%Y-%m-%dT%H:%M:%S")
```
But I'm not sure if the format is always the same else some variants could be added to the parse function. Besides that you can simply return a datetime object, no need to format it as a string.

0xAlcidius commented:
Thank you @bobkarreman for your comment and suggestion! I’ve noticed that Windows Scheduled Tasks can be a bit inconsistent when it comes to the StartBoundary. I did try using `datetime.strptime`, but given the ambiguity in the formats, I leaned towards string manipulation instead. Returning a datetime object would definitely be a cleaner approach, though.

Could you point me to the right place in the codebase to add the unit tests?

Schamper commented:
> Could you point me to the right place in the codebase to add the unit tests?

https://github.com/fox-it/dissect.target/blob/main/tests/plugins/os/windows/test_tasks.py

B0TAxy commented:
Should we also update the records and change the return type to datetime?

bobkarreman commented:
> Added a few test cases with timestamps observed in scheduled tasks. `str_to_isoformat` now also returns a datetime object as was suggested.

Please also include a test case for the case you described in the this PR ('2025-7-29T16:34:00')

bobkarreman commented:
> Thank you @bobkarreman for your comment and suggestion! I’ve noticed that Windows Scheduled Tasks can be a bit inconsistent when it comes to the StartBoundary. I did try using `datetime.strptime`, but given the ambiguity in the formats, I leaned towards string manipulation instead. Returning a datetime object would definitely be a cleaner approach.
> 
> Could you point me to the right place in the codebase to add the unit tests?

Parsing it with datetime.fromisoformat with some fallback logic is a cleaner way in my opinion (also fromisoformat has been improved in 3.11 so will work in almost all of the test cases).

Could be something like this:
```python
def parse(s):
    try:
        d = datetime.fromisoformat(s)
    except ValueError:
        s = s.replace(" ", "T")
        if 'Z' in s:
            d = datetime.strptime(s, "%Y-%m-%dT%H:%M:%SZ")
            return d.replace(tzinfo=timezone.utc)
        d = datetime.strptime(s, "%Y-%m-%dT%H:%M:%S")
    return d
```

0xAlcidius commented:
@bobkarreman Following your suggestion, I’ve refactored the function to make it cleaner while maintaining the same functionality. Additionally, I’ve added the missing test case you mentioned earlier.

yunzheng reviewed (changes requested):
Do you know if the timestamps in the tasks XML file are in UTC or local time?
